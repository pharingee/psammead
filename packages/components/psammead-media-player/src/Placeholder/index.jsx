import React from 'react';
import styled from 'styled-components';
import { string, func } from 'prop-types';
import Image from '@bbc/psammead-image';

const StyledPlaceholder = styled.div`
  cursor: pointer;
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
`;

const Placeholder = ({ onClick, src }) => (
  <StyledPlaceholder onClick={onClick}>
    <Image alt="Image Alt" src={src} />
  </StyledPlaceholder>
);

Placeholder.propTypes = {
  onClick: func.isRequired,
  src: string.isRequired,
};

export default Placeholder;
