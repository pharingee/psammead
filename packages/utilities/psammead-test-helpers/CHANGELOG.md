# Psammead Test Helpers Changelog

<!-- prettier-ignore -->
| Version | Description |
|---------|-------------|
| 3.0.2 | [PR#2166](https://github.com/bbc/psammead/pull/2166) remove async keyword from test helper to fix regenerator runtime plugin error |
| 3.0.1 | [PR#2123](https://github.com/bbc/psammead/pull/2125) Removes use of async/await |
| 3.0.0 | [PR#2104](https://github.com/bbc/psammead/pull/2104) add function to detect and render helmet for use in snapshots |
| 2.0.2 | [PR#1994](https://github.com/bbc/psammead/pull/1994) Add comment to clarify new shouldMatchSnapshot logic |
| 2.0.1 | [PR#1958](https://github.com/bbc/psammead/pull/1958) Update `shouldMatchSnapshot` to add firstChild logic |
| 2.0.0 | [PR#1917](https://github.com/bbc/psammead/pull/1917) Remove react-test-renderer and shallow rendering |
| 1.3.2 | [PR#1919](https://github.com/bbc/psammead/pull/1919) Accidental version bump |
| 1.3.1 | [PR#1849](https://github.com/bbc/psammead/pull/1849) Update README.md |
| 1.3.0 | [PR#1846](https://github.com/bbc/psammead/pull/1846/) Import test helper functions from simorgh |
| 1.2.1 | [PR#1803](https://github.com/bbc/psammead/pull/1803/) Patches broken links on badges in documentation |
| 1.2.0 | [PR#1794](https://github.com/bbc/psammead/pull/1794) Add david dependency badges |
| 1.1.1 | [PR#1682](https://github.com/bbc/psammead/pull/1682) Move all dev dependencies to top level package.json |
| 1.1.0   | [PR#1233](https://github.com/bbc/psammead/pull/1233) Add ESM modules entry |
| 1.0.2   | [PR#1083](https://github.com/bbc/psammead/pull/1083) Use `jest-styled-components@6.3.3` |
| 1.0.1   | [PR#732](https://github.com/bbc/psammead/pull/732) Use @testing-library/react for generating snapshots |
| 1.0.0   | [PR#679](https://github.com/bbc/psammead/pull/679) Bump version number |
| 0.3.4   | [PR#424](https://github.com/bbc/psammead/pull/424) Add Snyk badge to readme |
| 0.3.3   | [PR#323](https://github.com/BBC/psammead/pull/323) Update readme storybook badge |
| 0.3.2   | [PR#264](https://github.com/BBC/psammead/pull/319) Resolving package-lock issues. |
| 0.3.1   | [PR#264](https://github.com/BBC/psammead/pull/264) Resolving package-lock issues. |
| 0.3.0   | [PR#224](https://github.com/BBC-News/psammead/pull/224) Adds a helper method for testing package export values and uses it to test its own exports. |
| 0.2.1   | [PR#255](https://github.com/BBC-News/psammead/pull/255) Bumps package version to fix what appears to be an erroneously-published version. |
| 0.1.2   | [PR#212](https://github.com/BBC-News/psammead/pull/212) Update package description and README. |
| 0.1.1   | [PR#173](https://github.com/BBC-News/psammead/pull/173) Update PRs welcome link |
| 0.1.0   | [PR#80](https://github.com/BBC-News/psammead/pull/80) Create initial package, pulled in from [simorgh](https://github.com/BBC-News/simorgh). |
