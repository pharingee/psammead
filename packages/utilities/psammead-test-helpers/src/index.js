import { render } from '@testing-library/react';
import 'jest-styled-components';
import deepClone from 'ramda/src/clone';
import renderWithHelmet from './renderWithHelmet';

export const shouldMatchSnapshot = (title, component) => {
  it(title, done => {
    // select the first child to remove the pointless wrapping div from snapshots
    const removeWrappingDiv = container => container.firstChild;
    renderWithHelmet(component).then(({ container }) => {
      const hasOneChild = container.children.length === 1;
      /*
       * if the container has more than one child then it's a component that uses a
       * fragment at the top level so we should not select the first child because it
       * wouldn't snapshot the whole component
       */
      expect(
        hasOneChild ? removeWrappingDiv(container) : container,
      ).toMatchSnapshot();

      done();
    });
  });
};

export const isNull = (title, component) => {
  it(title, () => {
    const { container } = render(component);
    expect(container.firstChild).toBeNull();
  });
};

export const setWindowValue = (key, value) => {
  const windowValue = window[key];
  delete window[key];

  let newValue = value;

  if (value && typeof value === 'object') {
    newValue = {
      ...deepClone(windowValue),
      ...value,
    };
  }

  Object.defineProperty(window, key, {
    value: newValue,
    writable: true,
  });
};

export const resetWindowValue = (key, value) => {
  Object.defineProperty(window, key, {
    value,
    writable: true,
  });
};

export const suppressPropWarnings = warnings => {
  const { expectedWarnings } = window;
  if (expectedWarnings && Array.isArray(expectedWarnings)) {
    window.expectedWarnings = [...expectedWarnings, warnings];
  } else {
    window.expectedWarnings = [warnings];
  }
};

const errorIfMissingKey = (keys, object, message) => {
  keys.forEach(key => {
    if (!(key in object)) {
      throw new Error(`Missing value '${key}' in ${message}.`);
    }
  });
};

const checkKeysExistInBothObjects = (object1, object2, message1, message2) => {
  const object1Keys = Object.keys(object1);
  const object2Keys = Object.keys(object2);

  errorIfMissingKey(object1Keys, object2, message2);
  errorIfMissingKey(object2Keys, object1, message1);
};

const checkTypesOfExports = (
  actualExportsByName,
  actualExports,
  expectedExports,
  utilityName,
) => {
  actualExportsByName.forEach(actualExportName => {
    const actualExportValue = actualExports[utilityName][actualExportName];
    const expectedExport = expectedExports[utilityName][actualExportName];
    const typeCheck = typeof actualExportValue === expectedExport; // eslint-disable-line valid-typeof

    // if this fails it is likely that an export is missing from the unit test expectation
    expect(typeCheck).toBe(true);
  });
};

export const testUtilityPackages = (
  actualExports,
  expectedExports,
  packageName,
) => {
  const actualUtilities = Object.keys(actualExports);

  // check if the actual exported file has defined expected values to match and that the expected values are actually exported
  checkKeysExistInBothObjects(
    actualExports,
    expectedExports,
    `the actual utilities for '${packageName}'`,
    `the expected utilities for '${packageName}'`,
  );

  actualUtilities.forEach(utilityName => {
    const actualExportsByName = Object.keys(actualExports[utilityName]);

    // check if each of the actual exported consts have an expected value to match and that the expected values are actually exported
    checkKeysExistInBothObjects(
      actualExports[utilityName],
      expectedExports[utilityName],
      `the actual export for '${packageName}/${utilityName}'`,
      `the expected export for '${packageName}/${utilityName}'`,
    );

    checkTypesOfExports(
      actualExportsByName,
      actualExports,
      expectedExports,
      utilityName,
    );
  });
};
