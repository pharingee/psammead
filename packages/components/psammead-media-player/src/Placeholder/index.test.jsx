import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { shouldMatchSnapshot } from '@bbc/psammead-test-helpers';
import Placeholder from '.';

describe('Media Player: Placeholder', () => {
  const mockOnClick = jest.fn();

  afterEach(() => {
    jest.resetModules();
  });

  shouldMatchSnapshot(
    'should render a placeholder image with callback function',
    <Placeholder onClick={mockOnClick} src="http://foo.bar/placeholder.png" />,
  );

  it('should call onClick when clicked', () => {
    const { container } = render(
      <Placeholder
        onClick={mockOnClick}
        src="http://foo.bar/placeholder.png"
      />,
    );
    fireEvent.click(container.firstChild);
    expect(mockOnClick).toHaveBeenCalledTimes(1);
  });
});
