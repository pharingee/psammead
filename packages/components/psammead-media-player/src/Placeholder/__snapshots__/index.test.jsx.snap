// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Media Player: Placeholder should render a placeholder image with callback function 1`] = `
.c1 {
  display: block;
  width: 100%;
  visibility: visible;
}

.c0 {
  cursor: pointer;
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
}

<div
  class="c0"
>
  <img
    alt="Image Alt"
    class="c1"
    src="http://foo.bar/placeholder.png"
  />
</div>
`;
